django/contrib/auth/__init__.py

```diff
@@ -121,16 +121,22 @@ def login(request, user, backend=None):
     have to reauthenticate on every request. Note that data set during
     the anonymous session is retained when the user logs in.
     """
-    session_auth_hash = ''
+    session_auth_hash_sha256 = ''
+    session_auth_hash_sha1 = ''
     if user is None:
         user = request.user
     if hasattr(user, 'get_session_auth_hash'):
-        session_auth_hash = user.get_session_auth_hash()
+        session_auth_hash_sha256 = user.get_session_auth_hash(algo='sha256')
+        session_auth_hash_sha1 = user.get_session_auth_hash(algo='sha1')
 
     if SESSION_KEY in request.session:
         if _get_user_session_key(request) != user.pk or (
-                session_auth_hash and
-                not constant_time_compare(request.session.get(HASH_SESSION_KEY, ''), session_auth_hash)):
+            session_auth_hash_sha256 and
```

Comment:
Can you indent this and the next lines by 4 spaces as it was before, please. Also, have you considered storing the results of the sub-conditions into variables and comparing the variables in the `if` part:
```python
if SESSION_KEY in request.session:
    hash_session_key = request.session.get(HASH_SESSION_KEY, '')
    invalid_auth_hash_sha256 = session_auth_hash_sha256 and not constant_time_compare(hash_session_key, session_auth_hash_sha256)
    invalid_auth_hash_sha1 = session_auth_hash_sha1 and not constant_time_compare(hash_session_key, session_auth_hash_sha1)
    if _get_user_session_key(request) != user.pk or (invalid_auth_hash_sha256 and invalid_auth_hash_sha1):
        request.session.flush()
```